packages/lib-react-components/src/AnimatedNumber/AnimatedNumber.js

```diff
@@ -4,80 +4,94 @@ import { interpolate } from '@visx/vendor/d3-interpolate'
+ * @param {number} duration 
+ * @returns a Promise that resolves once the animation is complete.
+ */
+function animateValue(element, initialValue, value, duration) {
+  if (value === initialValue) {
+    return
   }
+  const interpolator = interpolate(initialValue, value)
+  return select(element)
+    .data([value])
+    .transition()
+    .duration(duration)
+    .textTween(() => t => formatValue(interpolator(t)))
+    .end()
+}
 
-  function lessAnimation() {
-    select(numRef.current)
+/**
+ * Use d3 to tween `element` from `initialValue` to `value`.
+ * @param {HTMLElement} element 
+ * @param {number} initialValue
+ * @param {number} value 
+ * @returns a Promise that resolves once the animation is complete.
+ */
+function lessAnimation(element, initialValue, value) {
+  if (value === initialValue) {
+    return
+  }
+  return select(element)
     .data([value])
     .transition()
     .duration(0)
-    .textTween(() => {
-      return () => {
-        setAnimated(true) // animation complete!
-        return formatValue(value)
-      }
-    })
-  }
+    .textTween(() => () => formatValue(value))
+    .end()
+}
 
-  function formatValue(num) {
-    return format(',d')(num)
-  }
+function AnimatedNumber({ duration = 1000, value }) {
+  const numRef = useRef(null)
+  const [displayedValue, setDisplayedValue] = useState(0)
 
   useEffect(() => {
-    // If we already animated the number once, don't observe intersection
-    // This could happen if the value prop updates, but page did not refresh
-    if (animated) return
-
+    /*
+     * We're animating the DOM node content outside of the React render cycle
+     * so use node.textContent to check if we've already rendered the value.
+     */
     const numElement = numRef.current
+    if (formatValue(value) === numElement.textContent) return
 
-    const intersectionObserver = new window.IntersectionObserver(entries => {
+    const intersectionObserver = new window.IntersectionObserver(async entries => {
       // If intersectionRatio is 0, the target is out of view and we do not need to do anything.
       if (entries[0].intersectionRatio <= 0) return
 
-      // Once target element is in viewport, animate it then unobserve
-      if (!prefersReducedMotion && !animated) {
-        animateValue()
-        intersectionObserver.unobserve(numElement)
-      } else {
-        lessAnimation()
-        intersectionObserver.unobserve(numElement)
-      }
+      /*
+       * Once target element is in viewport, animate it then unobserve.
+       * Only run the full animation if prefers reduced motion is false
+       * and the displayed value is 0.
+       */
+      const isFirstTime = numElement.textContent === '0'
+      const animationFunction =
+        !prefersReducedMotion() && isFirstTime ? animateValue : lessAnimation
+      intersectionObserver.unobserve(numElement)
+      await animationFunction(numElement, displayedValue, value, duration)
+      setDisplayedValue(value)
```

Comment:
The text content of `numElement` is `formatValue(value)` when the animation promise resolves. We need to update the value that React will render on the next render, even though it's exactly the same.